src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/catalog.cljs

```diff
@@ -0,0 +1,307 @@
+                                                     {:value v})])))
+        invalid-numbers (set (keep (fn [[k {:keys [value minimum maximum] :or {minimum -Infinity maximum Infinity}}]]
+                                     (when-not (and (= value (str (int value)))
+                                                    (<= minimum (int value) maximum))
+                                       k))
+                                   numeric-props))]
+    (when-not (empty? invalid-numbers)
+      {:error "Invalid number"
+       :invalid invalid-numbers})))
+
+
+(react/defc Questions
+  {:validate
+   (fn [{:keys [props state locals]}]
+     (let [{:keys [questions library-schema]} props
+           processed-attributes (->> (:attributes @state)
+                                     (utils/map-values (fn [val]
+                                                         (if (string? val)
+                                                           (not-empty (trim val))
+                                                           val)))
+                                     (utils/filter-values some?))
+           {:keys [error invalid]} (or (validate-required processed-attributes questions)
+                                       (validate-numbers processed-attributes library-schema))]
+       (swap! locals assoc :processed-attributes processed-attributes)
+       (when error
+         (swap! state assoc :invalid-properties invalid)
+         error)))
+   :get-attributes
+   (fn [{:keys [props locals]}]
+     (parse-attributes (:processed-attributes @locals) (:library-schema props)))
+   :get-initial-state
+   (fn [{:keys [props]}]
+     (let [{:keys [questions attributes library-schema]} props
+           prop->string (fn [prop] (if (seq? prop) (join ", " prop) (str prop)))
+           get-prop (fn [prop-key] (prop->string (get attributes prop-key
+                                                      (get-in library-schema [:properties prop-key :default]))))]
+       {:attributes
+        (reduce (fn [map prop-key] (assoc map prop-key (get-prop prop-key)))
+                {}
+                (map (comp keyword :property) questions))
+        :any-required? (some :required questions)}))
+   :render
+   (fn [{:keys [props state]}]
+     (let [{:keys [library-schema questions enumerate]} props]
+       [(if enumerate :ol :div) {}
+        (map
+          (fn [{:keys [property required inputHint renderHint]}]
+            (let [property-kwd (keyword property)
+                  {:keys [title type enum minimum consentCode]} (get-in library-schema [:properties property-kwd])
+                  error? (contains? (:invalid-properties @state) property-kwd)
+                  colorize (fn [style] (merge style (when error? {:borderColor (:exception-state style/colors)})))
+                  update-property #(swap! state update :attributes assoc property-kwd (.. % -target -value))]
+              [(if enumerate :li :div) {}
+               [:div {}
```

Comment:
nit: consider adding `:marginBottom "2px"` (or similar) here - if the div contains an `abbr` and is followed by a text input, the underline/bottom border of the abbr is hidden.